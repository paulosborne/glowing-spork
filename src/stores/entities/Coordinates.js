import { types } from 'mobx-state-tree';

const Coordinates = types
  .model('Coordinates', {
    longitude: types.maybeNull(types.number),
    latitude: types.maybeNull(types.number),
  })
  .views(self => ({
    get isValid() {
      return self.longitude !== null && self.latitude !== null;
    },
  }))
  .actions(self => ({
    setLongitude(value) {
      self.longitude = value;
    },
    setLatitude(value) {
      self.latitude = value;
    },
  }));

export default Coordinates;
