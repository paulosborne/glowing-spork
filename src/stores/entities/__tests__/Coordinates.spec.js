import { isStateTreeNode, getSnapshot } from 'mobx-state-tree';
import Coordinates from '../Coordinates';

describe('Coordinates', () => {
  let instance;

  beforeEach(() => {
    instance = Coordinates.create({});
  });

  it('should create an instance of Coordinates correctly', () => {
    expect(isStateTreeNode(instance)).toEqual(true);
    expect(getSnapshot(instance)).toMatchSnapshot();
  });

  describe('get isValid', () => {
    it('should return false by default', () => {
      expect(instance.isValid).toEqual(false);
    });

    it('should return true if the coordinates have been set', () => {
      instance.setLongitude(1);
      instance.setLatitude(52);
      expect(instance.isValid).toEqual(true);
    });
  });
});
