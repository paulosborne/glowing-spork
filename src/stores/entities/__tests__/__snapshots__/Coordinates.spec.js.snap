// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coordinates should create an instance of Coordinates correctly 1`] = `
Object {
  "latitude": null,
  "longitude": null,
}
`;
